bokehjs/src/lib/core/util/array.ts

```diff
@@ -104,7 +104,7 @@ export function range(start: number, stop?: number, step: number = 1): number[]
 }
 
 export function linspace(start: number, stop: number, num: number = 100): number[] {
-  const step = (stop - start) / (num - 1)
+  const step = num == 1 ? 0 : (stop - start) / (num - 1)
```

Comment:
This was the `linspace` bug when `num=1` as then you get a divide by zero for `step`, making it a `NaN`, which propagates through the BokehJS color conversion functions as a zero.